### PR TITLE
chore: Remove "From" add "FromName", "FromCode" in buildingBlocks models

### PR DESCRIPTION
**Pull request by RasmusGraabaek**
## Description

We have agreed, that the generic "From" method should be remove and we add "FromName"/"FromCode" instead

This should result in a better naming convention, or atleast force developer to use the "Right" method

### CODE REVIEW

File: source/BuildingBlocks.Domain/Models/Resolution.cs [hunk 1]
```diff
@@ -35,10 +35,21 @@ private Resolution(string name, string code)
...
+               ?? throw new InvalidOperationException($"{name} is not a valid {typeof(Currency)} {nameof(name)}");
+    }
+
+    public static Resolution FromCode(string code)
+    {
+        return GetAll<Resolution>()
+                   .First(
+                       resolution =>
+                           resolution.Code.Equals(code, StringComparison.OrdinalIgnoreCase))
+               ?? throw new InvalidOperationException($"{code} is not a valid {typeof(Currency)} {nameof(code)}");
```
Comment (RasmusGraabaek) [inline source/BuildingBlocks.Domain/Models/Resolution.cs:53]:
```suggestion
               ?? throw new InvalidOperationException($"{code} is not a valid {typeof(Resolution)} {nameof(code)}");
```

File: source/BuildingBlocks.Domain/Models/Resolution.cs [hunk 2]
```diff
@@ -35,10 +35,21 @@ private Resolution(string name, string code)
...
+    public static Resolution FromName(string name)
     {
-        return GetAll<Resolution>().First(resolution =>
-            resolution.Code.Equals(value, StringComparison.OrdinalIgnoreCase) ||
-            resolution.Name.Equals(value, StringComparison.OrdinalIgnoreCase));
+        return GetAll<Resolution>()
+                   .First(
+                       resolution =>
+                           resolution.Name.Equals(name, StringComparison.OrdinalIgnoreCase))
+               ?? throw new InvalidOperationException($"{name} is not a valid {typeof(Currency)} {nameof(name)}");
```
Comment (RasmusGraabaek) [inline source/BuildingBlocks.Domain/Models/Resolution.cs:44]:
```suggestion
               ?? throw new InvalidOperationException($"{name} is not a valid {typeof(Resolution)} {nameof(name)}");
```

File: source/OutgoingMessages.Application/MarketDocuments/NotifyAggregatedMeasureData/NotifyAggregatedMeasureDataXmlDocumentWriter.cs [hunk 3]
```diff
@@ -55,8 +55,13 @@ protected override async Task WriteMarketActivityRecordsAsync(IReadOnlyCollectio
...
             await WriteElementIfHasValueAsync("originalTransactionIDReference_Series.mRID", timeSeries.OriginalTransactionIdReference, writer).ConfigureAwait(false);
-            await writer.WriteElementStringAsync(DocumentDetails.Prefix, "marketEvaluationPoint.type", null, MeteringPointType.From(timeSeries.MeteringPointType).Code).ConfigureAwait(false);
-            await WriteElementIfHasValueAsync("marketEvaluationPoint.settlementMethod", timeSeries.SettlementType is null ? null : SettlementType.From(timeSeries.SettlementType).Code, writer).ConfigureAwait(false);
+            await writer.WriteElementStringAsync(
+                    DocumentDetails.Prefix,
+                    "marketEvaluationPoint.type",
+                    null,
+                    MeteringPointType.FromName(timeSeries.MeteringPointType).Code)
+                .ConfigureAwait(false);
+            await WriteElementIfHasValueAsync("marketEvaluationPoint.settlementMethod", timeSeries.SettlementType is null ? null : SettlementType.FromCode(timeSeries.SettlementType).Code, writer).ConfigureAwait(false);
```
Comment (RasmusGraabaek) [inline source/OutgoingMessages.Application/MarketDocuments/NotifyAggregatedMeasureData/NotifyAggregatedMeasureDataXmlDocumentWriter.cs:64]: Do we want stuff like this for readability: SettlementType.FromCode(timeSeries.SettlementType).Code

Or do we all know that SettlementType is the code?

File: source/Tests/Domain/Transactions/Aggregations/MeasurementUnitTests.cs [hunk 4]
```diff
@@ -22,9 +22,9 @@ public class MeasurementUnitTests
     [Theory]
     [InlineData("kwh")]
     [InlineData("KWH")]
-    public void Can_parse_from_name_or_code(string valueToParse)
+    public void Can_parse_from_code(string valueToParse)
```
Comment (RasmusGraabaek) [inline source/Tests/Domain/Transactions/Aggregations/MeasurementUnitTests.cs:25]: talk

File: source/BuildingBlocks.Domain/Models/BusinessReason.cs [hunk 5]
```diff
@@ -429,15 +429,21 @@ private BusinessReason(string name, string code)
 
     public static BusinessReason FromName(string name)
     {
-        var businessReason = GetAll<BusinessReason>().FirstOrDefault(br =>
-            br.Name.Equals(name, StringComparison.OrdinalIgnoreCase)) ?? throw new InvalidOperationException($"{name} is not a valid process type name");
-        return businessReason;
+        return GetAll<BusinessReason>()
+                   .First(
```
Comment (MWO1024) [inline source/BuildingBlocks.Domain/Models/BusinessReason.cs:433]: FoD?

File: source/BuildingBlocks.Domain/Models/BusinessReason.cs [hunk 6]
```diff
@@ -429,15 +429,21 @@ private BusinessReason(string name, string code)
...
+                   $"{name} is not a valid {typeof(BusinessReason)} {nameof(name)}");
     }
 
     public static BusinessReason FromCode(string code)
     {
-        var businessReason = GetAll<BusinessReason>().FirstOrDefault(br =>
-            br.Code.Equals(code, StringComparison.OrdinalIgnoreCase)) ?? throw new InvalidOperationException($"{code} is not a valid process type code");
-        return businessReason;
+        return GetAll<BusinessReason>()
+                   .First(
```
Comment (MWO1024) [inline source/BuildingBlocks.Domain/Models/BusinessReason.cs:443]: FoD?

File: source/BuildingBlocks.Domain/Models/DocumentFormat.cs [hunk 7]
```diff
@@ -28,9 +28,10 @@ private DocumentFormat(string name)
     {
     }
 
-    public static DocumentFormat From(string valueToParse)
+    public static DocumentFormat FromName(string name)
     {
         return GetAll<DocumentFormat>()
-            .First(format => format.Name.Equals(valueToParse, StringComparison.OrdinalIgnoreCase));
+                   .First(format => format.Name.Equals(name, StringComparison.OrdinalIgnoreCase))
```
Comment (MWO1024) [inline source/BuildingBlocks.Domain/Models/DocumentFormat.cs:34]: FoD?

**Review by MWO1024 — DISMISSED** [review]

File: source/Tests/BuildingBlocks/Models/ActorRoleTests.cs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        };
+
+        using var scope = new AssertionScope();
+        foreach (var test in actorRole)
+        {
+            ActorRole.FromName(test.Name).Should().Be(test.ExpectedValue);
+            ActorRole.FromCode(test.Code).Should().Be(test.ExpectedValue);
+        }
+
+        actorRole.Should().HaveCount(GetAllActorRoles().Count());
```
Comment (MWO1024) [inline source/Tests/BuildingBlocks/Models/ActorRoleTests.cs:59]: HaveSameCount()

File: source/Tests/BuildingBlocks/Models/BusinessReasonTests.cs [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        };
+
+        using var scope = new AssertionScope();
+        foreach (var test in businessReasons)
+        {
+            BusinessReason.FromName(test.Name).Should().Be(test.ExpectedValue);
+            BusinessReason.FromCode(test.Code).Should().Be(test.ExpectedValue);
+        }
+
+        businessReasons.Should().HaveCount(GetAllBusinessReasons().Count());
```
Comment (MWO1024) [inline source/Tests/BuildingBlocks/Models/BusinessReasonTests.cs:61]: HaveSameCount()

File: source/Tests/BuildingBlocks/Models/ChargeTypeTests.cs [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        };
+
+        using var scope = new AssertionScope();
+        foreach (var test in chargeTypes)
+        {
+            ChargeType.FromName(test.Name).Should().Be(test.ExpectedValue);
+            ChargeType.FromCode(test.Code).Should().Be(test.ExpectedValue);
+        }
+
+        chargeTypes.Should().HaveCount(GetAllChargeTypes().Count());
```
Comment (MWO1024) [inline source/Tests/BuildingBlocks/Models/ChargeTypeTests.cs:52]: HaveSameCount()

File: source/Tests/BuildingBlocks/Models/CurrencyTests.cs [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        };
+
+        using var scope = new AssertionScope();
+        foreach (var test in currencies)
+        {
+            Currency.FromName(test.Name).Should().Be(test.ExpectedValue);
+            Currency.FromCode(test.Code).Should().Be(test.ExpectedValue);
+        }
+
+        currencies.Should().HaveCount(GetAllCurrencies().Count());
```
Comment (MWO1024) [inline source/Tests/BuildingBlocks/Models/CurrencyTests.cs:50]: HaveSameCount() and so on for the remaining tests. I wonder if we can do something else... :worried: Count is fine and all, but does it really ensure we have the values we want? I can accidentally duplicate items in my list to make up for a new currency, without actually using the new currency.

**Review by MWO1024 — DISMISSED** [review]

**Review by MWO1024 — APPROVED** [review]

**Pull request merged**